src/relay.cpp

```diff
@@ -71,6 +71,153 @@ std::map<int, std::string> counterMap = {{DHCPv6_MESSAGE_TYPE_UNKNOWN, "Unknown"
 /* interface to vlan mapping */
 std::unordered_map<std::string, std::string> vlan_map;
 
+
+/* Class Options */
+
+bool Options::Add(OptionCode key, const uint8_t *value, uint16_t len) {
+    std::vector<uint8_t> option_v(value, value + len);
+    options[key] = option_v;
+    return true;
+}
+
+bool Options::Delete(OptionCode key) {
+    if (options.find(key) == options.end()) {
+        return false;
+    }
+    options.erase(key);
+    return true;
+}
+
+std::vector<uint8_t> Options::Get(OptionCode key) {
+    if (options.find(key) == options.end()) {
+        return std::vector<uint8_t>{};
+    }
+    return options[key];
+}
+
+std::vector<uint8_t> *Options::MarshalBinary() {
+    if (options.empty()) {
+        return nullptr;
+    }
+    list.clear();
+    for (auto &itr : options) {
+        uint16_t type = htons(itr.first);
+        uint16_t length = htons(itr.second.size());
+        list.push_back(((uint8_t *)&type)[0]);
+        list.push_back(((uint8_t *)&type)[1]);
+        list.push_back(((uint8_t *)&length)[0]);
+        list.push_back(((uint8_t *)&length)[1]);
+        list.insert(list.end(), itr.second.begin(), itr.second.end());
+    }
+    return &list;
+}
+
+bool Options::UnmarshalBinary(const uint8_t *packet, uint16_t length) {
+    while (length >= sizeof(dhcpv6_option)) {
```

Comment:
Should we log/warn if there is residue bytes that are less than an option size?

Reply:
Sure, I will update in next diff. Thanks for suggestion.